### PR TITLE
feat: Update assertion interface & Introduce composite assertion.

### PR DESCRIPTION
**Pull request by ryan-gang**

### CODE REVIEW

File: internal/test_cases/send_command_test_case.go [hunk 1]
```diff
@@ -45,10 +47,16 @@ func (t SendCommandTestCase) Run(client *resp_client.RespConnection, logger *log
 		}
 	}
 
-	if err = t.Assertion.Run(value); err != nil {
-		return err
+	result := t.Assertion.Run(value)
+	if result.IsFailure() {
+		if result.SuccessMessages != nil {
+			logger.Successf(strings.Join(result.SuccessMessages, "\n"))
+		}
```
Comment (ryan-gang) [inline internal/test_cases/send_command_test_case.go:54]: Even if we have a failure, we are printing out any success logs if present.
I have only added this in a single file. 
Looking for your feedback on this approach.

**Comment by rohitpaulk** [comment]
@ryan-gang failing tests here too

**Comment by rohitpaulk** [comment]
@ryan-gang I'll look at this once you have fixtures in place

**Comment by ryan-gang** [comment]
Updated fixtures @rohitpaulk.

**Comment by rohitpaulk** [comment]
Hmm, this feels a bit overkill - almost like we're exploding the interface just to satisfy one specific use-case. Lemme think about this a bit

**Comment by rohitpaulk** [comment]
Closing in favour of https://github.com/codecrafters-io/redis-tester/pull/116

**Pull request closed without merging**